### PR TITLE
Tests of reprocessing submissions that yielded entity.error earlier

### PR DESCRIPTION
**Pull request by ktuite**
Related to https://github.com/getodk/central/issues/547

The main code change here is that in the list of submissions to reprocess when toggling the dataset `approvalRequired` flag, submissions that were already processed for some reason and led to an entity processing `entity.error` are now filtered out and not reprocessed.

Two main scenarios: 
1. A submission failed to create an entity, and then the status was changed so there was a new `submission.update` event. Because the first attempt created an error, this will no longer be re-processed.
2. A submission failed to update an entity, and then something with the dataset changed (maybe a non-existent entity came into existence) but that submission wont be reprocessed in the "reprocess all unprocessed submissions" flow. (It was awkward it got caught up in this process anyway, but we don't have a good way of knowing ahead of time if a submission is about creating or updating so we have to re-examine it.)
    * In this entity update case, if you _edit_ the submission, that can re-trigger the desired entity update.

#### Before submitting this PR, please make sure you have:

- [ ] run `make test-full` and confirmed all checks still pass OR confirm CircleCI build passes
- [ ] verified that any code from external sources are properly credited in comments or that everything is internally sourced

### CODE REVIEW

File: lib/model/query/datasets.js
Comment (matthew-white) [inline lib/model/query/datasets.js:460]: Should we use `submissionDefId` instead of `submissionId`? If the submission had an error, then an edit, and no error specifically associated with the edit, it should still be processed I think.

Comment (ktuite) [reply lib/model/query/datasets.js:460]: I thought about this, but the way this would happen is
- submission
- approval
- error
- edit (new sub def)
- DON'T approve yet
- change dataset flag

... and that seems so specialized to have the auto processing happen then. 

Someone could approve earlier or could do the edit after changing the dataset flag and it should work.

It was easier for me to think about how if the submission had ever caused an error with an entity, it wouldn't get caught up in this reprocessing scenario.

File: test/integration/api/datasets.js
Comment (ktuite) [inline test/integration/api/datasets.js:3588]: processed for the first time

**Review by matthew-white — APPROVED** [review]

File: lib/model/query/datasets.js
Comment (ktuite) [reply lib/model/query/datasets.js]: Changing the code so it _will_ process any new version of a submission that hasn't specifically generated an error.

**Pull request merged**
